File: src/main/java/com/ansupercomputer/supercomputerhub/virtualmachine/VirtualMachineController.java
package com.ansupercomputer.supercomputerhub.virtualmachine;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping(value = "vm")
public class VirtualMachineController {
    private final VirtualMachineService service;

    @Autowired
    public VirtualMachineController(VirtualMachineService service) {
        this.service = service;
    }

    /**
     * Returns a list of all virtual machines
     */
    @GetMapping
    public List<VirtualMachine> getVirtualMachines() {
        return service.getAllVirtualMachines();
    }

    /**
     * Add a virtual machine
     */
    @PostMapping
    public void addVirtualMachine(@RequestBody VirtualMachine vm) {
        service.addVirtualMachine(vm);
    }

    /**
     * Delete a virtual machine
     */
    @DeleteMapping(value = "{id}/delete")
    public void deleteVirtualMachine(@PathVariable int id) {
        service.deleteVirtualMachine(id);
    }

    /**
     * Get a specific virtual machine by id
     */
    @GetMapping(value = "{id}")
    public VirtualMachine getVirtualMachineById(@PathVariable int id) {
        return service.getVirtualMachineById(id);
    }
}

File: src/main/java/com/ansupercomputer/supercomputerhub/virtualmachine/VirtualMachineController.kt
package com.ansupercomputer.supercomputerhub.virtualmachine

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["vm"])
class VirtualMachineController (@Autowired val service: VirtualMachineService) {
    /**
     * Returns a list of all virtual machines
     */
    @GetMapping
    fun getVirtualMachines(): List<VirtualMachine> {
        return service.getAllVirtualMachines()
    }

    /**
     * Add a virtual machine
      */
    @PostMapping
    fun addVirtualMachine(@RequestBody vm: VirtualMachine) {
        return service.addVirtualMachine(vm)
    }

    /**
     * Delete a virtual machine
     */
    @DeleteMapping(value = ["{id}/delete"])
    fun deleteVirtualMachine(@PathVariable id: Int): Boolean {
        return service.deleteVirtualMachine(id)
    }

    /**
     * Get a specific virtual machine by id
     */
    @GetMapping(value = ["{id}"])
    fun getVirtualMachineById(@PathVariable id: Int): VirtualMachine? {
        return service.getVirtualMachineById(id)
    }

}
